### ISSUE TITLE
nginx: add listen option

### ISSUE THREAD

**Issue by apeyroux** [NixOS/nixpkgs#27287]
## Issue description

I have a server that has multiple IPs. The HTTP/S server should be able to listen only on a specific IP. (Other ip must also be able to listen 80 or 443).

In a "normal" configuration I will add:
```
listen 195.154.xxx.xxx:80; 
listen 195.154.xxx.xxx:443;
```
In the "modules/services/web-servers/nginx/default.nix" file ( https://github.com/NixOS/nixpkgs/blob/02129a8788423cb0b3dd4248c1ae342866d0d7ad/nixos/modules/services/web-servers/nginx/default.nix#L129 ) there is a section with a variable "listenString". This variable is not "overloadable" via the configuration.

Can we add a listen option that takes a list of ip to listen to? Example:

`services.nginx.virtualHosts.<name>.listen = [ "192.xxx.xxx.xxx:80", "192.xxx.xxx.xxx:443 ..."] `

Nginx doc : http://nginx.org/en/docs/http/ngx_http_core_module.html#listen

**Comment by rnhmjoj** [comment]
Because of `.forceSSL` the address and port have to be different variables. Something like this maybe could do it:
```
services.nginx.virtualHosts.<name>.listen = [ 
  { address = "192.xxx.xxx.xxx"; port = 80; }
  { address = "192.xxx.xxx.xxx"; port = 443; }
];
```

**Comment by rnhmjoj** [comment]
Ok, It's not so trivial. What should `forceSSL` do when there are multiple addresses and ports?

**Comment by fpletz** [comment]
Fixed by #27403

**Comment by apeyroux** [comment]
Thanks
